src/home_robot_spot/home_robot_spot/spot_client.py

```diff
@@ -473,11 +473,24 @@ def gps(self):
 
     @property
     def compass(self):
-        return self.raw_observations["base_xyt"][0][0][2]
+        return self.raw_observations["base_xyt"][0][1][0]
 
     @property
-    def position(self):
-        return self.raw_observations["base_xyt"][0][0][:3]
+    def current_relative_position(self):
+        xy = self.gps
+        compass = self.compass
+        relative_gps = xy - self.start_gps
+        relative_gps = self.rot_compass @ relative_gps
```

Comment:
Instead we can use sophus as we are with Stretch - this would give me some more confidence